recipe-client-addon/lib/NormandyApi.jsm

```diff
@@ -49,8 +49,29 @@ this.NormandyApi = {
     return this.apiCall("post", endpoint, data);
   },
 
+  apiIndex() {
+    if (!indexPromise) {
+      indexPromise = this.get(prefs.getCharPref("api_url")).then(res => res.json);
+    }
+    return indexPromise;
+  },
+
+  makeApiUrl(endpoint) {
+    let base = prefs.getCharPref("api_url");
+    // Remove trailing slash if present
+    if (base.slice(-1) === "/") {
+      base = base.slice(0, -1);
+    }
+    // Remove leading slash, if present
+    if (endpoint[0] === "/") {
+      endpoint = endpoint.slice(1);
+    }
+    return `${base}/${endpoint}`;
+  },
+
   fetchRecipes: Task.async(function* (filters = {}) {
-    const recipeResponse = yield this.get("recipe/signed/", filters);
+    const signedRecipeUrls = (yield this.apiIndex())["recipes-signed"];
+    const recipeResponse = yield this.get(signedRecipeUrls, filters, {enabled: true});
```

Comment:
The API endpoint `/recipe/` is the listing page for all objects of type `recipe`. `recipe/$id/` would  be an individual recipe. `recipe/signed` is a specialization of the listing view that also lists many recipes, and includes signatures.

The singular form of this would include an id or a slug specifying an individual object.

That being said, these names aren't right. It should be `signedRecipesUrl` and `recipesResponse`, most likely. I'll fix that.